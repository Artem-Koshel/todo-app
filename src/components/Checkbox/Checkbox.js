import React from 'react';
import { PropTypes } from 'prop-types';
import styles from './Checkbox.module.scss';
import { ReactComponent as Checkmark } from './checkmark.svg';

export default function Checkbox({ id, text, checked, onChange }) {
  return (
    <label className={styles.checkbox} htmlFor={id}>
      <div className={`${styles.styledCheckbox} ${checked ? styles.checked : styles.unchecked}`}>
        {checked && <Checkmark width="15px" height="15px" className={styles.checkmark} />}
      </div>
      <input id={id} type="checkbox" checked={checked} onChange={onChange} />
      <span className={`${styles.itemText} ${checked ? styles.crossed : undefined}`}>{text}</span>
    </label>
  );
}

Checkbox.propTypes = {
  id: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  text: PropTypes.string.isRequired,
  checked: PropTypes.bool.isRequired,
  onChange: PropTypes.func.isRequired,
};
