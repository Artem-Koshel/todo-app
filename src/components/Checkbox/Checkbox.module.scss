.checkbox {
    display: flex;
    cursor: pointer;

    input[type="checkbox"] {
        display: none;
    }

    .styledCheckbox {
        width: 25px;
        height: 25px;
        box-sizing: border-box;
        border-radius: 50%;
        position: relative;

        .checkmark {
            position: absolute;
            inset: 0;
            margin: auto;
        }
    }

    .checked {
        background: linear-gradient(135deg,#7fbef4,#a473f0);
        background-size: cover;
    }

    .unchecked {
        border: 1px solid lightgrey;
    }
}

.itemText {
    margin-left: 15px;
    line-height: 25px;
     
    &.crossed {
        color: lightgray;
        text-decoration: line-through;
    }
}

