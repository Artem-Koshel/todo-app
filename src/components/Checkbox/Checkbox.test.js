import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Checkbox from './Checkbox';

describe('Checkbox component', () => {
  it.each([true, false])('renders correctly, checked: %p', (checked) => {
    const { container } = render(
      <Checkbox id="testId" text="label text" checked={checked} onChange={jest.fn()} />
    );
    expect(container).toMatchSnapshot();
  });

  it('exec callback on checked/unchecked', async () => {
    const user = userEvent.setup();
    const callback = jest.fn();
    render(<Checkbox id="testId" text="label text" checked={true} onChange={callback} />);
    await user.click(screen.getByText('label text'));
    expect(callback).toBeCalled();
  });
});
