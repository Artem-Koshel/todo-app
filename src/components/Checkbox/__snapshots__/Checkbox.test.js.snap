// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox component renders correctly, checked: false 1`] = `
<div>
  <label
    class="checkbox"
    for="testId"
  >
    <div
      class="styledCheckbox unchecked"
    />
    <input
      id="testId"
      type="checkbox"
    />
    <span
      class="itemText undefined"
    >
      label text
    </span>
  </label>
</div>
`;

exports[`Checkbox component renders correctly, checked: true 1`] = `
<div>
  <label
    class="checkbox"
    for="testId"
  >
    <div
      class="styledCheckbox checked"
    >
      <svg
        class="checkmark"
        height="15px"
        width="15px"
      >
        checkmark.svg
      </svg>
    </div>
    <input
      checked=""
      id="testId"
      type="checkbox"
    />
    <span
      class="itemText crossed"
    >
      label text
    </span>
  </label>
</div>
`;
